### PR TITLE
feat: Decline problem report

### PR DESCRIPTION
**Pull request by Tommylans**
Implementation for the decline problem report in the Paradym wallet.

### CODE REVIEW

File: packages/agent/src/hooks/useAcceptDidCommPresentation.ts [hunk 1]
```diff
@@ -157,9 +157,33 @@ export function useAcceptDidCommPresentation(proofExchangeId: string) {
...
+        .pipe(
+          // Correct record with id and state
+          filter(
+            (event) =>
+              event.payload.proofRecord.id === proofExchangeId &&
+              [ProofState.Declined].includes(event.payload.proofRecord.state)
+          ),
+          // 10 seconds to complete exchange
+          timeout(10000),
+          first()
```
Comment (TimoGlastra) [inline packages/agent/src/hooks/useAcceptDidCommPresentation.ts:174]: I think the method call will wait with returning until this state is reached, so i think we can remove it

**Review by TimoGlastra — APPROVED** [review]
Nice! How about credential decline?

File: packages/agent/src/hooks/useAcceptDidCommPresentation.ts (hunk 1 above)
Comment (Tommylans) [reply packages/agent/src/hooks/useAcceptDidCommPresentation.ts:174]: But it's quite nice to have this functionality right just like the accept action. And the state is updated locally in afj right for the record or does it have to wait for the external party?

**Comment by Tommylans** [comment]
> Nice! How about credential decline?

Yes, added the credential decline now 😄

**Pull request merged**